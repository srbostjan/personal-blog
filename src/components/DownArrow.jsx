import React, { useState, useEffect } from 'react'

const DownArrow = () => {
  const [scroll, setScroll] = useState(0);
  useEffect(() => {
    window.addEventListener('scroll', handleScroll);
  });
  let handleScroll = () => {
    setScroll(window.scrollY);
  };
  let handleClick = (e) => {
    if(scroll > 200) {
      window.scrollTo(0, -window.innerHeight);
      e.target.classList.remove('up');
    } else {
      window.scrollTo(0, window.innerHeight);
      e.target.classList.add('up');
    }
  };
  return (
    <a onClick={handleClick}><i className="fas fa-arrow-circle-right down-arrow"></i></a>
  );
};

export default DownArrow;