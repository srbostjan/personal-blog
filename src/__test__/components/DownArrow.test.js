import React from 'react';
import { mount } from 'enzyme';
import { create } from 'react-test-renderer';
import DownArrow from '../../components/DownArrow';

describe('<DownArrow />', () => {
  test('Render componente DownArrow', () => {
    const component = mount(<DownArrow />);
    expect(component.length).toEqual(1);
  });
});
describe('DownArrow Snapshot', () => {
  test('DownArrow Snapshot', () => {
    const component = create(<DownArrow />);
    expect(component.toJSON()).toMatchSnapshot();
  });
})
