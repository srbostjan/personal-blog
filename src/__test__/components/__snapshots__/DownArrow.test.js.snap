// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DownArrow Snapshot DownArrow Snapshot 1`] = `
<button
  onClick={[Function]}
  type="button"
>
  <i
    className="fas fa-arrow-circle-right down-arrow"
  />
</button>
`;
